crates/sncast/src/starknet_commands/verification/base.rs

```diff
@@ -0,0 +1,90 @@
+use anyhow::{anyhow, Context, Result};
+use async_trait::async_trait;
+use camino::Utf8PathBuf;
+use reqwest::StatusCode;
+use serde::Serialize;
+use sncast::response::structs::VerifyResponse;
+use sncast::Network;
+use starknet::core::types::FieldElement;
+use std::ffi::OsStr;
+use walkdir::WalkDir;
+
+#[async_trait]
+pub trait VerificationInterface {
+    fn new(network: Network, workspace_dir: Utf8PathBuf) -> Self;
+    async fn verify(
+        &self,
+        contract_address: Option<FieldElement>,
+        class_hash: Option<FieldElement>,
+        class_name: String,
+    ) -> Result<VerifyResponse>;
+    fn gen_explorer_url(&self) -> Result<String>;
```

Comment:
1. Similarly to `verify`, implementations of this method differ only with the url source. How about introducing a new trait method, let's say `get_base_url`, implementing it manually on both types and using in a default implementation of `gen_explorer_url`?
2. This method never fails. Are there any reasons for it to return a `Result`?